### ISSUE TITLE
Updating taxon position not work properly

### ISSUE THREAD

**Issue by bigboss86** [Sylius/Sylius#9909]
**Sylius version affected**: 1.3.3

**Description**  
Using leaf nodes up and down not position the taxon correctly

**Steps to reproduce**  
Create taxons and use the leaf nodes.

**Possible Solution**  
Not investigated yet.

![captura de pantalla de 2018-11-08 14-54-22](https://user-images.githubusercontent.com/5973407/48217475-36c74b00-e366-11e8-8479-0d6606ba741b.png)

**Comment by pamil** [comment]
Thank you for the bug report! We'll investigate more into that, it would be a great help if anyone would like to try to solve this one.

**Comment by smashou** [comment]
I think it's because the position field for the taxon is not set properly. When you want to move the leaf nodes it just increment or decrement the position 1 by 1. But sometimes you can have taxon position like this one:
```
- taxon_1 (parent_id NULL, position 12)
--- taxon_1.1 (parent_id taxon_1, position 2)
--  taxon_1.2 (parent_id taxon_1, position 3)
--  taxon_1.3 (parent_id taxon_1, position 8)
```
So if you push the up arrow for the taxon_1.3 the position will be decremented by 1 and it will be = position 7. But it's not enough for it to go above the  taxon_1.2.

I hope it's clear enough to help you.

**Comment by bigboss86** [comment]
Hi @smashou. Thanks for your reply.
In the example of Sylius demo, when i do click in arrow up in Books taxon, this move it to bottom of the list. so, i think don't work properly as you says. Maybe the position of taxons in Sylius is wrong too?

**Comment by jacquesbh** [comment]
Do not stale?

**Comment by jacquesbh** [comment]
Do not stale

**Comment by igormukhingmailcom** [comment]
The solution:
- Create `sylius_admin_ajax_taxon_move_up` and `sylius_admin_ajax_taxon_move_down` actions rather than `sylius_admin_ajax_taxon_move`
- We shouldn't send position as that an break things, so `taxonMoveUp` / `taxonMoveDown` / `TaxonPositionType` should be removed
- `sylius_admin_ajax_taxon_move_up` should switch positions between current taxon and taxon with max position (but less than current taxon's position) with same parent as current taxon
- `sylius_admin_ajax_taxon_move_down` should switch positions between current taxon and taxon with min position (but greater than current taxon's position) with same parent as current taxon

**Comment by igormukhingmailcom** [comment]
@pamil , could you take a look into #11905 please?

**Comment by igormukhingmailcom** [comment]
@AdamKasp , this issue seems like fixed by #11905
